shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -220,6 +274,44 @@ private void setTextInputEditingState(View view, TextInputChannel.TextEditState
     }
 
     private void clearTextInputClient() {
-        mClient = 0;
+        if (inputTarget.type == InputTarget.Type.PLATFORM_VIEW) {
+            // Focus changes in the framework tree have no guarantees on the order focus nodes are notified. A node
+            // that lost focus may be notified before or after a node that gained focus.
+            // When moving the focus from a Flutter text field to an AndroidView, it is possible that the Flutter text
+            // field's focus node will be notified that it lost focus after the AndroidView was notified that it gained
+            // focus. When this happens the text field will send a clearTextInput command which we ignore.
+            // By doing this we prevent the framework from clearing a platform view input client(the only way to do so
+            // is to set a new framework text client). I don't see an obvious use case for "clearing" a platform views
+            // text input client, and it may be error prone as we don't know how the platform view manages the input
+            // connection and we probably shouldn't interfere.
+            // If we ever want to allow the framework to clear a platform view text client we should probably consider
+            // changing the focus manager such that focus nodes that lost focus are notified before focus nodes that
+            // gained focus as part of the same focus event.
+            return;
+        }
+        inputTarget = new InputTarget(InputTarget.Type.NO_TARGET, 0);
+        unlockPlatformViewInputConnection();
+    }
+
+    static private class InputTarget {
+        enum Type {
+            NO_TARGET,
+            // InputConnection is managed by the TextInputPlugin, and events are forwarded to the Flutter framework.
+            FRAMEWORK_CLIENT,
+            // InputConnection is managed by an embedded platform view.
+            PLATFORM_VIEW
+        }
+
+        public InputTarget(Type type, int id) {
```

Comment:
Can you add a nullability annotation to `type`?

Reply:
done